### PR TITLE
Enable requireSpread rule.

### PR DESCRIPTION
**Pull request by xtian**
https://github.com/jscs-dev/node-jscs/blob/master/lib/rules/require-spread.js

### CODE REVIEW

**Comment by rwjblue** [comment]
:+1: - I like this requirement. 

Can you please add these to the good examples list:

```
var Blah = Ember.Object.extend({
  init() {
    this._super(...arguments); // normal usage

    // odd, but still needed in some cases
    var _super = this._super;
    // some async here
    _super.apply(this, arguments);
  }
});

function bar() {
  // I expect `this` to be a specific thing
}

bar.apply(someSpecialSnowflake, argsArray);
```

**Comment by hzoo** [comment]
Probably a few bugs or other cases it could catch I didn't get around to doing - https://github.com/jscs-dev/node-jscs/issues/1482

**Comment by xtian** [comment]
@rwjblue Alright I added those. Let me know if you'd like any other changes. Thanks!

File: tests/fixtures/rules/require-spread/bad/example.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+const wrap = (f, g) => (...args) => g.apply(g, [f].concat(args));
+instance.method.apply(instance, args);
```
Comment (rwjblue) [inline tests/fixtures/rules/require-spread/bad/example.js]: Can you move this second example into a separate file? If we don't do that the various types of bad things are guaranteed to fail...

**Comment by xtian** [comment]
@rwjblue: Fixed, sorry!

**Comment by rwjblue** [comment]
No worries! Looks like Travis is angry though.

**Comment by xtian** [comment]
Yeah it seems as though the second example doesn't actually fail even though it is listed as an invalid example in jscs.

**Comment by rwjblue** [comment]
Glad we caught it :stuck_out_tongue_winking_eye:, maybe you could report it to JSCS?

**Comment by rwjblue** [comment]
@xtian - Did you report this issue upstream?

**Comment by xtian** [comment]
@rwjblue I mentioned it on jscs-dev/node-jscs#1482 as the issue seems related

**Comment by brzpegasus** [comment]
Going to merge this, despite https://github.com/jscs-dev/node-jscs/issues/1482 not being fixed. Having the rule in here means we'll have partial coverage at least (instead of none at all). And when that issue does get fixed, it should be pulled in automatically when users install `ember-suave`.

**Pull request merged**